boards: nordic: Add initial support for nRF54LM20A/ns

degjorva opened the pull request:
Add board files for nRF54LM20A/ns.
Update existing nRF54LM20A board files to support this.

degjorva committed:
boards: nordic: Add initial support for nRF54LM20A/ns

Add board files for nRF54LM20A/ns.
Update existing nRF54LM20A board files to support this.

Signed-off-by: Dag Erik Gjørvad <dag.erik.gjorvad@nordicsemi.no>

degjorva commented on boards/nordic/nrf54lm20dk/nrf54lm20dk_nrf54lm20a_cpuapp_ns.yaml:
```diff
@@ -0,0 +1,22 @@
+toolchain:
+  - gnuarmemb
+  - zephyr
+ram: 256
+flash: 968
```
Oh good catch, the flash is an old value

tomi-font commented on boards/nordic/nrf54lm20dk/Kconfig.defconfig:
```diff
@@ -1,9 +1,21 @@
 # Copyright (c) 2025 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
```
unused
```suggestion
```

tomi-font commented on boards/nordic/nrf54lm20dk/nrf54lm20dk_nrf54lm20a_cpuapp_ns.dts:
```diff
@@ -0,0 +1,68 @@
+/ {
+	/*
+	 * Default SRAM planning when building for nRF54LM20A with ARM TrustZone-M support
+	 * - Lowest 208 kB SRAM allocated to Secure image (sram0_s).
+	 * - Upper 208 kB SRAM allocated to Non-Secure image (sram0_ns).
```
This doesn't match the actual sizes (256K). Fix either?

tomi-font commented on boards/nordic/nrf54lm20dk/nrf54lm20dk_nrf54lm20a_cpuapp_ns.dts:
```diff
@@ -0,0 +1,75 @@
+};
+
+/* Get a node label for wi-fi spi to use in shield files */
+wifi_spi: &spi22 {};
```
Maybe move it from `nrf54lm20dk_nrf54lm20a_cpuapp.dts` to `nrf54lm20dk_nrf54lm20a-common.dtsi`?

tomi-font commented on dts/vendor/nordic/nrf54lm20a_ns_partition.dtsi:
```diff
@@ -0,0 +1,62 @@
+	 * Default NVM layout on NRF54LM20A Application MCU without BL2:
+	 * This layout matches (by necessity) that in the TF-M repository:
+	 *
+	 * 0x0000_0000 Secure image primary (384 KB)
```
```suggestion
	 * 0x0000_0000 Secure image primary (512 KB)
```
?

tomi-font commented on dts/vendor/nordic/nrf54lm20a_ns_partition.dtsi:
```diff
@@ -0,0 +1,62 @@
+	 * 0x0008_0000 Protected Storage Area (16 KB)
+	 * 0x0008_4000 Internal Trusted Storage Area (16 KB)
+	 * 0x0008_8000 OTP / NV counters area (8 KB)
+	 * 0x0008_A000 Non-secure image primary (1524 KB)
```
1356 or 1524? this is inconsistent

tomi-font replied:
1356 or 1524?
```suggestion
	 * 0x0008_A000 Non-secure image primary (1356 KB)
```

nordicjm commented on boards/nordic/nrf54lm20dk/nrf54lm20a_cpuapp_common.dtsi:
```diff
@@ -58,43 +58,7 @@
-		};
-	};
-};
+/* TF-M partitions are defined in the NS board DTS file */
```
comment not needed and doesn't even apply to secure board target

nordicjm commented on boards/nordic/nrf54lm20dk/nrf54lm20dk_nrf54lm20a_cpuapp_ns.dts:
```diff
@@ -0,0 +1,68 @@
+	};
+};
+
+&bt_hci_controller {
+	status = "disabled";
+};
+
+&uart30 {
+	/* Disable so that TF-M can use this UART */
+	status = "disabled";
+};
```
why can't these definitions be moved from the common file to the normal cpuapp file so that they don't need to be disabled here?

nordicjm commented on dts/vendor/nordic/nrf54lm20a_ns_partition.dtsi:
```diff
@@ -0,0 +1,62 @@
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* nRF54LM20A has 2036 kB of non-volatile memory (RRAM) but the last
+		 * 96 kB are reserved for the FLPR MCU.
```
no, no no no, we agreed not to do this again @carlescufi

nordicjm replied:
no flpr reservation in normal builds without flpr, use full RRAM for application core

nordicjm commented on dts/vendor/nordic/nrf54lm20a_partition.dtsi:
```diff
@@ -0,0 +1,37 @@
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* nRF54LM20A has 2036 kB of non-volatile memory (RRAM) but the last
+		 * 96 kB are reserved for the FLPR MCU, so we have ~1940 kB available.
+		 */
```
as above, no

nordicjm replied:
total size of partitions below: 1696KiB, according to comment above, there is 1940 available, so where has the other 244 magically vanished off to? The original partitioning came to a total size of 1896 so something is very wrong here

degjorva committed:
manifest: Update commit id for trusted-firmware-m

Update commit id for trusted-firmware-m
to bring in support for nRF54LM20A/ns.

Signed-off-by: Dag Erik Gjørvad <dag.erik.gjorvad@nordicsemi.no>

nordicjm commented on boards/nordic/nrf54lm20dk/nrf54lm20dk_nrf54lm20a_cpuapp_ns.dts:
```diff
@@ -0,0 +1,68 @@
+			/* Non-Secure image memory */
+			reg = <0x20034000 DT_SIZE_K(208)>;
+		};
+	};
```
ironically right this moment as part of something else it's been decided that this is wrong and an abuse of the `reserved-memory` feature but it will be reworked in future

tomi-font reviewed (dismissed):
@d3zd3z please review (or unassign)

kartben commented:
> @d3zd3z please review (or unassign)

re-assigning to nRF platform maintainers.
